test/test.js

```diff
@@ -761,6 +761,52 @@ test( "elementValue() finds radios/checkboxes only within the current form", fun
 	ok( !v.elementValue( foreignRadio ) );
 } );
 
+test( "elementValue() returns the file input's name without the prefix 'C:\\fakepath\\' ", function() {
+	var v = $( "#userForm" ).validate(),
+		fileInput = {
+			name: "fakeFile",
+			type: "file",
+			files: {},
+			nodeName: "INPUT",
+			value: "C:\\fakepath\\somefile.txt",
+			form: $( "#userForm" )[ 0 ],
+			hasAttribute: function() { return false; }
+		},
+		oldRules = $.fn.rules,
+		oldvalidationTargetFor = v.validationTargetFor;
+
+	equal( v.elementValue( fileInput ), "somefile.txt" );
+
+	$.fn.rules = function() {
+		return {
+			minlength: 10
+		};
+	};
+
+	v.validationTargetFor = function() {
+		return fileInput;
+	};
+
+	ok( v.check( fileInput ), "The fake file input is valid (length = 12, minlength = 10)" );
+
+	fileInput.value = "C:\\fakepath\\file.txt";
+	ok( !v.check( fileInput ), "The fake file input is invalid (length = 8, minlength = 10)" );
+
+	$.fn.rules = function() {
+		return {
+			maxlength: 10
+		};
+	};
+
+	ok( v.check( fileInput ), "The fake file input is valid (length = 8, maxlength = 10)" );
+
+	fileInput.value = "C:\\fakepath\\fakefile.txt";
+	ok( !v.check( fileInput ), "The fake file input is invalid (length = 12, maxlength = 10)" );
+
+	$.fn.rules = oldRules;
```

Comment:
> I dont remember other unit tests doing such cleanup, so I am pretty sure we would have a lot of problems in case qunit would not already clean those things up (or boots a fresh environment for each test)

I don't see such thing in their documentation (If I didn't miss something).
I guess we should ask someone from the core team for an answer, I don't know if jzaefferer has the time to do so.